sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -662,9 +680,18 @@ private void setColdStart(final @Nullable Bundle savedInstanceState) {
   }
 
   private void finishAppStartSpan() {
-    final @Nullable SentryDate appStartEndTime = AppStartState.getInstance().getAppStartEndTime();
+    final @NotNull TimeSpan appStartTimeSpan = getAppStartTimeSpan();
+    final @Nullable SentryDate appStartEndTime = appStartTimeSpan.getProjectedStopTimestamp();
     if (performanceEnabled && appStartEndTime != null) {
       finishSpan(appStartSpan, appStartEndTime);
     }
   }
+
+  private @NotNull TimeSpan getAppStartTimeSpan() {
+    final @NotNull TimeSpan appStartTimeSpan =
+        options.isEnablePerformanceV2()
```

Comment:
I think we should also check for the API version > 24 and return the legacy timespan in this case, right? Otherwise if performanceV2 is used on lower API versions, the `getAppStartTimeSpan` will be always not started